extensions/podman/packages/extension/scripts/podman-download.ts

```diff
@@ -452,7 +256,10 @@ export class Podman5DownloadMachineOS {
     // grab applehv as annotations / disktype
     const keepManifests = manifests.filter(manifest => {
       const annotations = manifest.annotations;
-      return annotations && annotations.disktype === 'applehv';
+      return (
+        annotations &&
+        ((this.#wsl && annotations.disktype === 'wsl') || (!this.#wsl && annotations.disktype === 'applehv'))
```

Comment:
probably use an enum rather than a flag wsl

if we need to add the support of Hyper-V, it's another item of the enum, else here you'll be stuck with the !wsl as it could be both Hyper-V or applehv